examples/quantum_walk.ipynb

```diff
@@ -0,0 +1,381 @@
+       "          0.00000000e+00],\n",
+       "       [  0.00000000e+00,   0.00000000e+00,   0.00000000e+00,\n",
+       "          0.00000000e+00]])"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "had = np.sqrt(1/2) * np.array([[1, 1], [1, -1]])\n",
+    "pauli_x = np.array([[0, 1], [1, 0]])\n",
+    "Q = np.kron(had, had)\n",
+    "Q.conj().T.dot(hamil).dot(Q)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The time evolution operator $e^{-iHt}$ is also diagonalized by the same transformation. In particular we have\n",
+    "\n",
+    "$$\n",
+    "Q^\\dagger e^{-iHt}Q = \\begin{pmatrix} \n",
+    "            e^{-i4t} & 0 & 0 & 0 \\\\\n",
+    "            0 & 1 & 0 & 0 \\\\\n",
+    "            0 & 0 & 1 & 0 \\\\\n",
+    "            0 & 0 & 0 & 1\n",
+    "           \\end{pmatrix}\n",
+    "$$\n",
+    "\n",
+    "Which is just a [CPHASE00](http://pyquil.readthedocs.io/en/latest/gates.html) gate with an angle of $-4t$. The circuit to simulate these is then extremely simple:\n",
+    "\n",
+    "![](circuit.png \"Circuit for $K_4$ CTQW\")\n",
+    "\n",
+    "(Taken from paper above, $\\gamma = 1$ in our case.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pyquil.quil as pq\n",
+    "import pyquil.api as api\n",
+    "from pyquil.gates import H, X, CPHASE00"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "qvm = api.SyncConnection()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def k_4_ctqw(t):\n",
+    "    p = pq.Program()\n",
+    "    \n",
+    "    #   Change to diagonal basis\n",
+    "    p.inst(H(0))\n",
+    "    p.inst(H(1))\n",
+    "    p.inst(X(0))\n",
+    "    p.inst(X(1))\n",
+    "    \n",
+    "    #    Time evolve\n",
+    "    p.inst(CPHASE00(-4*t, 0, 1))\n",
+    "    \n",
+    "    #   Change back to computational basis\n",
+    "    p.inst(X(0))\n",
+    "    p.inst(X(1))\n",
+    "    p.inst(H(0))\n",
+    "    p.inst(H(1))\n",
+    "    \n",
+    "    return p"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's compare the quantum walk with a classical random walk. The classical time evolution operator is $e^{-(\\mathcal{T} - I) t}$ where $\\mathcal{T}$ is the transition matrix of the graph.\n",
+    "\n",
+    "We choose as our initial condition $\\left|\\psi(0)\\right> = \\left|0\\right>$, that is the walker starts on the first node. Therefore, due to symmetry, the probability of occupation of all nodes besides $\\left| 0 \\right>$ is the same. "
```

Comment:
@andrenmelo This seems to have rendering problems. At least the GH NB viewer does not parse this with MathJax.

Reply:
Hmm, strange, it's rendering fine locally. I'll look into it.